docs/proposals/20240102-upgrade-yurtappset-to-v1beta1.md

```diff
@@ -0,0 +1,258 @@
+|              title              | authors   | reviewers | creation-date | last-updated | status |
+| :-----------------------------: | --------- | --------- | ------------- | ------------ | ------ |
+| Upgrade yurt-app-set to v1beta1 | @luc99hen |           | 2024-01-02    |              |        |
+
+# Upgrade yurt-app-set to v1beta1
+
+<!-- TOC -->
+* [Upgrade yurt-app-set to v1beta1](#upgrade-yurtappset-to-v1beta1)
+	* [Summary](#summary)
+	* [Motivation](#motivation)
+		* [Goals](#goals)
+		* [Non-Goals/Future Work](#non-goalsfuture-work)
+	* [Proposal](#proposal)
+	* [Implementation History](#implementation-history)
+<!-- TOC -->
+
+## Summary
+
+Due to the multi-region nature of openyurt, we introduced a series of crds to faciliate the management of multi-region workloads. yurtappset and yurtappdaemon are used for workload distribution and management; and yurtappoverrider takes over the ability to differentiate configurations across multiple geographies. Although they serve different scenarios and functions, much of the functionality overlaps, and combining them into a single component helps reduce user perception and usage costs. Therefore, we combine the ability of the three crds into yurtappset v1beta1.
+
+## Motivation
+
+1. One crd instead of three is more user-friendly to users 
+2. In the previous technical route, the rendering of the workload came from both yurtappset and yurtappoverrider, and the version management of the workload needed to take into account the changes from both parts, which was very complex.
+
+
+### Goals
+
+1. Introduce YurtAppSet v1beta1 to replace YurtAppSet v1alpha1, YurtAppDaemon and YurtAppOverrider
+
```

Comment:
Workloads are dynamically managed in response to changes in nodepools, this means when new nodepools are established and meet the NodePoolSelector in YurtAppSet, workloads can be automatically created and assigned to them. Conversely, when existing nodepools no longer fulfill the required NodePoolSelector, associated workloads will be removed accordingly.

Reply:
done